static/app/views/dashboardsV2/detail.tsx

```diff
@@ -435,11 +436,14 @@ class DashboardDetail extends Component<Props, State> {
               }
 
               if (organization.features.includes('dashboard-grid-layout')) {
-                this.saveLayoutWithNewWidgets(
+                const newLayout = this.saveLayoutWithNewWidgets(
                   organization.id,
                   newDashboard.id,
                   newDashboard.widgets
                 );
+                // Unset modifiedDashboard and set newLayout at the same time
+                // to render newLayout with updated dashboard prop
+                this.setState({layout: newLayout, modifiedDashboard: null});
```

Comment:
I think the PR looks fine, though starting on line 454 there is also another `setState`. Might be worth combining these two `setState` calls in a future PR to make things cleaner

Reply:
How do you feel about:
```js
let newLayout;
if (<grid layout feature flag>) {
  newLayout = saveDashboardLayout(...)
  // remove setState
}
...
this.setState({
  ...,
  layout: newLayout ?? layout
});
```